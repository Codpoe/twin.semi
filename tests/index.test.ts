import { expect, test } from 'vitest';
import { twinSemiPreset } from 'twin.semi';

test('preset', () => {
  expect(twinSemiPreset()).toMatchObject({
    theme: {
      extend: {
        colors: {},
      },
    },
  });

  expect(twinSemiPreset()).toMatchSnapshot();

  expect(
    twinSemiPreset({
      keyFormatter: key => (key.startsWith('--semi-') ? key.substring(7) : key),
    })
  ).toMatchSnapshot();
});
