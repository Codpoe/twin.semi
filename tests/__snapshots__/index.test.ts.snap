// Vitest Snapshot v1

exports[`preset 1`] = `
{
  "theme": {
    "extend": {
      "colors": {
        "semi-amber-0": [Function],
        "semi-amber-1": [Function],
        "semi-amber-2": [Function],
        "semi-amber-3": [Function],
        "semi-amber-4": [Function],
        "semi-amber-5": [Function],
        "semi-amber-6": [Function],
        "semi-amber-7": [Function],
        "semi-amber-8": [Function],
        "semi-amber-9": [Function],
        "semi-black": [Function],
        "semi-blue-0": [Function],
        "semi-blue-1": [Function],
        "semi-blue-2": [Function],
        "semi-blue-3": [Function],
        "semi-blue-4": [Function],
        "semi-blue-5": [Function],
        "semi-blue-6": [Function],
        "semi-blue-7": [Function],
        "semi-blue-8": [Function],
        "semi-blue-9": [Function],
        "semi-color-bg-0": "var(--semi-color-bg-0)",
        "semi-color-bg-1": "var(--semi-color-bg-1)",
        "semi-color-bg-2": "var(--semi-color-bg-2)",
        "semi-color-bg-3": "var(--semi-color-bg-3)",
        "semi-color-bg-4": "var(--semi-color-bg-4)",
        "semi-color-black": "var(--semi-color-black)",
        "semi-color-border": "var(--semi-color-border)",
        "semi-color-danger": "var(--semi-color-danger)",
        "semi-color-danger-active": "var(--semi-color-danger-active)",
        "semi-color-danger-hover": "var(--semi-color-danger-hover)",
        "semi-color-danger-light-active": "var(--semi-color-danger-light-active)",
        "semi-color-danger-light-default": "var(--semi-color-danger-light-default)",
        "semi-color-danger-light-hover": "var(--semi-color-danger-light-hover)",
        "semi-color-default": "var(--semi-color-default)",
        "semi-color-default-active": "var(--semi-color-default-active)",
        "semi-color-default-hover": "var(--semi-color-default-hover)",
        "semi-color-disabled-bg": "var(--semi-color-disabled-bg)",
        "semi-color-disabled-border": "var(--semi-color-disabled-border)",
        "semi-color-disabled-fill": "var(--semi-color-disabled-fill)",
        "semi-color-disabled-text": "var(--semi-color-disabled-text)",
        "semi-color-fill-0": "var(--semi-color-fill-0)",
        "semi-color-fill-1": "var(--semi-color-fill-1)",
        "semi-color-fill-2": "var(--semi-color-fill-2)",
        "semi-color-focus-border": "var(--semi-color-focus-border)",
        "semi-color-highlight": "var(--semi-color-highlight)",
        "semi-color-highlight-bg": "var(--semi-color-highlight-bg)",
        "semi-color-info": "var(--semi-color-info)",
        "semi-color-info-active": "var(--semi-color-info-active)",
        "semi-color-info-disabled": "var(--semi-color-info-disabled)",
        "semi-color-info-hover": "var(--semi-color-info-hover)",
        "semi-color-info-light-active": "var(--semi-color-info-light-active)",
        "semi-color-info-light-default": "var(--semi-color-info-light-default)",
        "semi-color-info-light-hover": "var(--semi-color-info-light-hover)",
        "semi-color-link": "var(--semi-color-link)",
        "semi-color-link-active": "var(--semi-color-link-active)",
        "semi-color-link-hover": "var(--semi-color-link-hover)",
        "semi-color-link-visited": "var(--semi-color-link-visited)",
        "semi-color-nav-bg": "var(--semi-color-nav-bg)",
        "semi-color-overlay-bg": "var(--semi-color-overlay-bg)",
        "semi-color-primary": "var(--semi-color-primary)",
        "semi-color-primary-active": "var(--semi-color-primary-active)",
        "semi-color-primary-disabled": "var(--semi-color-primary-disabled)",
        "semi-color-primary-hover": "var(--semi-color-primary-hover)",
        "semi-color-primary-light-active": "var(--semi-color-primary-light-active)",
        "semi-color-primary-light-default": "var(--semi-color-primary-light-default)",
        "semi-color-primary-light-hover": "var(--semi-color-primary-light-hover)",
        "semi-color-secondary": "var(--semi-color-secondary)",
        "semi-color-secondary-active": "var(--semi-color-secondary-active)",
        "semi-color-secondary-disabled": "var(--semi-color-secondary-disabled)",
        "semi-color-secondary-hover": "var(--semi-color-secondary-hover)",
        "semi-color-secondary-light-active": "var(--semi-color-secondary-light-active)",
        "semi-color-secondary-light-default": "var(--semi-color-secondary-light-default)",
        "semi-color-secondary-light-hover": "var(--semi-color-secondary-light-hover)",
        "semi-color-shadow": "var(--semi-color-shadow)",
        "semi-color-success": "var(--semi-color-success)",
        "semi-color-success-active": "var(--semi-color-success-active)",
        "semi-color-success-disabled": "var(--semi-color-success-disabled)",
        "semi-color-success-hover": "var(--semi-color-success-hover)",
        "semi-color-success-light-active": "var(--semi-color-success-light-active)",
        "semi-color-success-light-default": "var(--semi-color-success-light-default)",
        "semi-color-success-light-hover": "var(--semi-color-success-light-hover)",
        "semi-color-tertiary": "var(--semi-color-tertiary)",
        "semi-color-tertiary-active": "var(--semi-color-tertiary-active)",
        "semi-color-tertiary-hover": "var(--semi-color-tertiary-hover)",
        "semi-color-tertiary-light-active": "var(--semi-color-tertiary-light-active)",
        "semi-color-tertiary-light-default": "var(--semi-color-tertiary-light-default)",
        "semi-color-tertiary-light-hover": "var(--semi-color-tertiary-light-hover)",
        "semi-color-text-0": "var(--semi-color-text-0)",
        "semi-color-text-1": "var(--semi-color-text-1)",
        "semi-color-text-2": "var(--semi-color-text-2)",
        "semi-color-text-3": "var(--semi-color-text-3)",
        "semi-color-warning": "var(--semi-color-warning)",
        "semi-color-warning-active": "var(--semi-color-warning-active)",
        "semi-color-warning-hover": "var(--semi-color-warning-hover)",
        "semi-color-warning-light-active": "var(--semi-color-warning-light-active)",
        "semi-color-warning-light-default": "var(--semi-color-warning-light-default)",
        "semi-color-warning-light-hover": "var(--semi-color-warning-light-hover)",
        "semi-color-white": "var(--semi-color-white)",
        "semi-cyan-0": [Function],
        "semi-cyan-1": [Function],
        "semi-cyan-2": [Function],
        "semi-cyan-3": [Function],
        "semi-cyan-4": [Function],
        "semi-cyan-5": [Function],
        "semi-cyan-6": [Function],
        "semi-cyan-7": [Function],
        "semi-cyan-8": [Function],
        "semi-cyan-9": [Function],
        "semi-green-0": [Function],
        "semi-green-1": [Function],
        "semi-green-2": [Function],
        "semi-green-3": [Function],
        "semi-green-4": [Function],
        "semi-green-5": [Function],
        "semi-green-6": [Function],
        "semi-green-7": [Function],
        "semi-green-8": [Function],
        "semi-green-9": [Function],
        "semi-grey-0": [Function],
        "semi-grey-1": [Function],
        "semi-grey-2": [Function],
        "semi-grey-3": [Function],
        "semi-grey-4": [Function],
        "semi-grey-5": [Function],
        "semi-grey-6": [Function],
        "semi-grey-7": [Function],
        "semi-grey-8": [Function],
        "semi-grey-9": [Function],
        "semi-indigo-0": [Function],
        "semi-indigo-1": [Function],
        "semi-indigo-2": [Function],
        "semi-indigo-3": [Function],
        "semi-indigo-4": [Function],
        "semi-indigo-5": [Function],
        "semi-indigo-6": [Function],
        "semi-indigo-7": [Function],
        "semi-indigo-8": [Function],
        "semi-indigo-9": [Function],
        "semi-light-blue-0": [Function],
        "semi-light-blue-1": [Function],
        "semi-light-blue-2": [Function],
        "semi-light-blue-3": [Function],
        "semi-light-blue-4": [Function],
        "semi-light-blue-5": [Function],
        "semi-light-blue-6": [Function],
        "semi-light-blue-7": [Function],
        "semi-light-blue-8": [Function],
        "semi-light-blue-9": [Function],
        "semi-light-green-0": [Function],
        "semi-light-green-1": [Function],
        "semi-light-green-2": [Function],
        "semi-light-green-3": [Function],
        "semi-light-green-4": [Function],
        "semi-light-green-5": [Function],
        "semi-light-green-6": [Function],
        "semi-light-green-7": [Function],
        "semi-light-green-8": [Function],
        "semi-light-green-9": [Function],
        "semi-lime-0": [Function],
        "semi-lime-1": [Function],
        "semi-lime-2": [Function],
        "semi-lime-3": [Function],
        "semi-lime-4": [Function],
        "semi-lime-5": [Function],
        "semi-lime-6": [Function],
        "semi-lime-7": [Function],
        "semi-lime-8": [Function],
        "semi-lime-9": [Function],
        "semi-orange-0": [Function],
        "semi-orange-1": [Function],
        "semi-orange-2": [Function],
        "semi-orange-3": [Function],
        "semi-orange-4": [Function],
        "semi-orange-5": [Function],
        "semi-orange-6": [Function],
        "semi-orange-7": [Function],
        "semi-orange-8": [Function],
        "semi-orange-9": [Function],
        "semi-pink-0": [Function],
        "semi-pink-1": [Function],
        "semi-pink-2": [Function],
        "semi-pink-3": [Function],
        "semi-pink-4": [Function],
        "semi-pink-5": [Function],
        "semi-pink-6": [Function],
        "semi-pink-7": [Function],
        "semi-pink-8": [Function],
        "semi-pink-9": [Function],
        "semi-purple-0": [Function],
        "semi-purple-1": [Function],
        "semi-purple-2": [Function],
        "semi-purple-3": [Function],
        "semi-purple-4": [Function],
        "semi-purple-5": [Function],
        "semi-purple-6": [Function],
        "semi-purple-7": [Function],
        "semi-purple-8": [Function],
        "semi-purple-9": [Function],
        "semi-red-0": [Function],
        "semi-red-1": [Function],
        "semi-red-2": [Function],
        "semi-red-3": [Function],
        "semi-red-4": [Function],
        "semi-red-5": [Function],
        "semi-red-6": [Function],
        "semi-red-7": [Function],
        "semi-red-8": [Function],
        "semi-red-9": [Function],
        "semi-teal-0": [Function],
        "semi-teal-1": [Function],
        "semi-teal-2": [Function],
        "semi-teal-3": [Function],
        "semi-teal-4": [Function],
        "semi-teal-5": [Function],
        "semi-teal-6": [Function],
        "semi-teal-7": [Function],
        "semi-teal-8": [Function],
        "semi-teal-9": [Function],
        "semi-violet-0": [Function],
        "semi-violet-1": [Function],
        "semi-violet-2": [Function],
        "semi-violet-3": [Function],
        "semi-violet-4": [Function],
        "semi-violet-5": [Function],
        "semi-violet-6": [Function],
        "semi-violet-7": [Function],
        "semi-violet-8": [Function],
        "semi-violet-9": [Function],
        "semi-white": [Function],
        "semi-yellow-0": [Function],
        "semi-yellow-1": [Function],
        "semi-yellow-2": [Function],
        "semi-yellow-3": [Function],
        "semi-yellow-4": [Function],
        "semi-yellow-5": [Function],
        "semi-yellow-6": [Function],
        "semi-yellow-7": [Function],
        "semi-yellow-8": [Function],
        "semi-yellow-9": [Function],
      },
    },
  },
}
`;

exports[`preset 2`] = `
{
  "theme": {
    "extend": {
      "colors": {
        "amber-0": [Function],
        "amber-1": [Function],
        "amber-2": [Function],
        "amber-3": [Function],
        "amber-4": [Function],
        "amber-5": [Function],
        "amber-6": [Function],
        "amber-7": [Function],
        "amber-8": [Function],
        "amber-9": [Function],
        "black": [Function],
        "blue-0": [Function],
        "blue-1": [Function],
        "blue-2": [Function],
        "blue-3": [Function],
        "blue-4": [Function],
        "blue-5": [Function],
        "blue-6": [Function],
        "blue-7": [Function],
        "blue-8": [Function],
        "blue-9": [Function],
        "color-bg-0": "var(--semi-color-bg-0)",
        "color-bg-1": "var(--semi-color-bg-1)",
        "color-bg-2": "var(--semi-color-bg-2)",
        "color-bg-3": "var(--semi-color-bg-3)",
        "color-bg-4": "var(--semi-color-bg-4)",
        "color-black": "var(--semi-color-black)",
        "color-border": "var(--semi-color-border)",
        "color-danger": "var(--semi-color-danger)",
        "color-danger-active": "var(--semi-color-danger-active)",
        "color-danger-hover": "var(--semi-color-danger-hover)",
        "color-danger-light-active": "var(--semi-color-danger-light-active)",
        "color-danger-light-default": "var(--semi-color-danger-light-default)",
        "color-danger-light-hover": "var(--semi-color-danger-light-hover)",
        "color-default": "var(--semi-color-default)",
        "color-default-active": "var(--semi-color-default-active)",
        "color-default-hover": "var(--semi-color-default-hover)",
        "color-disabled-bg": "var(--semi-color-disabled-bg)",
        "color-disabled-border": "var(--semi-color-disabled-border)",
        "color-disabled-fill": "var(--semi-color-disabled-fill)",
        "color-disabled-text": "var(--semi-color-disabled-text)",
        "color-fill-0": "var(--semi-color-fill-0)",
        "color-fill-1": "var(--semi-color-fill-1)",
        "color-fill-2": "var(--semi-color-fill-2)",
        "color-focus-border": "var(--semi-color-focus-border)",
        "color-highlight": "var(--semi-color-highlight)",
        "color-highlight-bg": "var(--semi-color-highlight-bg)",
        "color-info": "var(--semi-color-info)",
        "color-info-active": "var(--semi-color-info-active)",
        "color-info-disabled": "var(--semi-color-info-disabled)",
        "color-info-hover": "var(--semi-color-info-hover)",
        "color-info-light-active": "var(--semi-color-info-light-active)",
        "color-info-light-default": "var(--semi-color-info-light-default)",
        "color-info-light-hover": "var(--semi-color-info-light-hover)",
        "color-link": "var(--semi-color-link)",
        "color-link-active": "var(--semi-color-link-active)",
        "color-link-hover": "var(--semi-color-link-hover)",
        "color-link-visited": "var(--semi-color-link-visited)",
        "color-nav-bg": "var(--semi-color-nav-bg)",
        "color-overlay-bg": "var(--semi-color-overlay-bg)",
        "color-primary": "var(--semi-color-primary)",
        "color-primary-active": "var(--semi-color-primary-active)",
        "color-primary-disabled": "var(--semi-color-primary-disabled)",
        "color-primary-hover": "var(--semi-color-primary-hover)",
        "color-primary-light-active": "var(--semi-color-primary-light-active)",
        "color-primary-light-default": "var(--semi-color-primary-light-default)",
        "color-primary-light-hover": "var(--semi-color-primary-light-hover)",
        "color-secondary": "var(--semi-color-secondary)",
        "color-secondary-active": "var(--semi-color-secondary-active)",
        "color-secondary-disabled": "var(--semi-color-secondary-disabled)",
        "color-secondary-hover": "var(--semi-color-secondary-hover)",
        "color-secondary-light-active": "var(--semi-color-secondary-light-active)",
        "color-secondary-light-default": "var(--semi-color-secondary-light-default)",
        "color-secondary-light-hover": "var(--semi-color-secondary-light-hover)",
        "color-shadow": "var(--semi-color-shadow)",
        "color-success": "var(--semi-color-success)",
        "color-success-active": "var(--semi-color-success-active)",
        "color-success-disabled": "var(--semi-color-success-disabled)",
        "color-success-hover": "var(--semi-color-success-hover)",
        "color-success-light-active": "var(--semi-color-success-light-active)",
        "color-success-light-default": "var(--semi-color-success-light-default)",
        "color-success-light-hover": "var(--semi-color-success-light-hover)",
        "color-tertiary": "var(--semi-color-tertiary)",
        "color-tertiary-active": "var(--semi-color-tertiary-active)",
        "color-tertiary-hover": "var(--semi-color-tertiary-hover)",
        "color-tertiary-light-active": "var(--semi-color-tertiary-light-active)",
        "color-tertiary-light-default": "var(--semi-color-tertiary-light-default)",
        "color-tertiary-light-hover": "var(--semi-color-tertiary-light-hover)",
        "color-text-0": "var(--semi-color-text-0)",
        "color-text-1": "var(--semi-color-text-1)",
        "color-text-2": "var(--semi-color-text-2)",
        "color-text-3": "var(--semi-color-text-3)",
        "color-warning": "var(--semi-color-warning)",
        "color-warning-active": "var(--semi-color-warning-active)",
        "color-warning-hover": "var(--semi-color-warning-hover)",
        "color-warning-light-active": "var(--semi-color-warning-light-active)",
        "color-warning-light-default": "var(--semi-color-warning-light-default)",
        "color-warning-light-hover": "var(--semi-color-warning-light-hover)",
        "color-white": "var(--semi-color-white)",
        "cyan-0": [Function],
        "cyan-1": [Function],
        "cyan-2": [Function],
        "cyan-3": [Function],
        "cyan-4": [Function],
        "cyan-5": [Function],
        "cyan-6": [Function],
        "cyan-7": [Function],
        "cyan-8": [Function],
        "cyan-9": [Function],
        "green-0": [Function],
        "green-1": [Function],
        "green-2": [Function],
        "green-3": [Function],
        "green-4": [Function],
        "green-5": [Function],
        "green-6": [Function],
        "green-7": [Function],
        "green-8": [Function],
        "green-9": [Function],
        "grey-0": [Function],
        "grey-1": [Function],
        "grey-2": [Function],
        "grey-3": [Function],
        "grey-4": [Function],
        "grey-5": [Function],
        "grey-6": [Function],
        "grey-7": [Function],
        "grey-8": [Function],
        "grey-9": [Function],
        "indigo-0": [Function],
        "indigo-1": [Function],
        "indigo-2": [Function],
        "indigo-3": [Function],
        "indigo-4": [Function],
        "indigo-5": [Function],
        "indigo-6": [Function],
        "indigo-7": [Function],
        "indigo-8": [Function],
        "indigo-9": [Function],
        "light-blue-0": [Function],
        "light-blue-1": [Function],
        "light-blue-2": [Function],
        "light-blue-3": [Function],
        "light-blue-4": [Function],
        "light-blue-5": [Function],
        "light-blue-6": [Function],
        "light-blue-7": [Function],
        "light-blue-8": [Function],
        "light-blue-9": [Function],
        "light-green-0": [Function],
        "light-green-1": [Function],
        "light-green-2": [Function],
        "light-green-3": [Function],
        "light-green-4": [Function],
        "light-green-5": [Function],
        "light-green-6": [Function],
        "light-green-7": [Function],
        "light-green-8": [Function],
        "light-green-9": [Function],
        "lime-0": [Function],
        "lime-1": [Function],
        "lime-2": [Function],
        "lime-3": [Function],
        "lime-4": [Function],
        "lime-5": [Function],
        "lime-6": [Function],
        "lime-7": [Function],
        "lime-8": [Function],
        "lime-9": [Function],
        "orange-0": [Function],
        "orange-1": [Function],
        "orange-2": [Function],
        "orange-3": [Function],
        "orange-4": [Function],
        "orange-5": [Function],
        "orange-6": [Function],
        "orange-7": [Function],
        "orange-8": [Function],
        "orange-9": [Function],
        "pink-0": [Function],
        "pink-1": [Function],
        "pink-2": [Function],
        "pink-3": [Function],
        "pink-4": [Function],
        "pink-5": [Function],
        "pink-6": [Function],
        "pink-7": [Function],
        "pink-8": [Function],
        "pink-9": [Function],
        "purple-0": [Function],
        "purple-1": [Function],
        "purple-2": [Function],
        "purple-3": [Function],
        "purple-4": [Function],
        "purple-5": [Function],
        "purple-6": [Function],
        "purple-7": [Function],
        "purple-8": [Function],
        "purple-9": [Function],
        "red-0": [Function],
        "red-1": [Function],
        "red-2": [Function],
        "red-3": [Function],
        "red-4": [Function],
        "red-5": [Function],
        "red-6": [Function],
        "red-7": [Function],
        "red-8": [Function],
        "red-9": [Function],
        "teal-0": [Function],
        "teal-1": [Function],
        "teal-2": [Function],
        "teal-3": [Function],
        "teal-4": [Function],
        "teal-5": [Function],
        "teal-6": [Function],
        "teal-7": [Function],
        "teal-8": [Function],
        "teal-9": [Function],
        "violet-0": [Function],
        "violet-1": [Function],
        "violet-2": [Function],
        "violet-3": [Function],
        "violet-4": [Function],
        "violet-5": [Function],
        "violet-6": [Function],
        "violet-7": [Function],
        "violet-8": [Function],
        "violet-9": [Function],
        "white": [Function],
        "yellow-0": [Function],
        "yellow-1": [Function],
        "yellow-2": [Function],
        "yellow-3": [Function],
        "yellow-4": [Function],
        "yellow-5": [Function],
        "yellow-6": [Function],
        "yellow-7": [Function],
        "yellow-8": [Function],
        "yellow-9": [Function],
      },
    },
  },
}
`;
